src/services/product.services.test.ts

```diff
@@ -0,0 +1,246 @@
+
+      const productDeleted = await ProductService.deleteProductService(productCreated!.id);
+
+      expect(findSpy).toHaveBeenCalledTimes(1);
+      expect(findSpy).toHaveBeenCalledWith({ where: { id: productCreated!.id } });
+
+      expect(deleteSpy).toHaveBeenCalledTimes(1);
+      expect(deleteSpy).toHaveBeenCalledWith({ where: { id: productCreated!.id } });
+
+      expect(productDeleted).toBeDefined();
+      expect(productDeleted).toEqual(productCreated);
+    });
+
+    it('should serialize product response', async () => {
+      jest.spyOn(db.product, 'delete').mockResolvedValue(productRandomRaw);
+
+      const productSerialized: Product = {
+        id: productRandomRaw.id,
+        title: productRandomRaw.title,
+        description: productRandomRaw.description,
+        userId: productRandomRaw.userId,
+      };
+
+      expect(productSerialized).toEqual(ProductSerializer.serialize(productRandomRaw));
+    });
+
+    it('should throw an ApiError (NOT_FOUND) when product is not found', async () => {
+      jest.spyOn(db.product, 'findUnique').mockResolvedValue(null);
+
+      const productId = faker.number.int();
+      await expect(ProductService.deleteProductService(productId)).rejects.toThrow(new ApiError(errors.NOT_FOUND));
+    });
+
+    it('should throw an Error on database error', async () => {
+      jest.spyOn(db.product, 'findUnique').mockResolvedValue(productRandomRaw);
+      jest.spyOn(db.product, 'delete').mockRejectedValue(new Error('Database error'));
+
+      const productId = faker.number.int();
+      await expect(ProductService.deleteProductService(productId)).rejects.toThrow(new Error('Database error'));
+    });
+  });
+
+  describe('Update Product', () => {
+    it('should call findUnique and update product with the correct arguments and return the product updated', async () => {
+      const productCreated = await createProduct({ userId });
+
+      const findSpy = jest.spyOn(db.product, 'findUnique');
+      const updateSpy = jest.spyOn(db.product, 'update');
+
+      const { id } = productCreated;
+
+      const productEdited: UpdateProductRequest = {
+        title: productCreated.title + '111',
+        description: productCreated.description + '111',
+      };
+
+      const productUpdated = await ProductService.updateProductService(id, productEdited);
+
+      expect(findSpy).toHaveBeenCalledTimes(1);
+      expect(findSpy).toHaveBeenCalledWith({ where: { id } });
+
+      expect(updateSpy).toHaveBeenCalledTimes(1);
+      expect(updateSpy).toHaveBeenCalledWith({ data: { ...productEdited }, where: { id } });
+
+      expect(productUpdated).toBeDefined();
+
+      expect(productUpdated).toEqual(
+        expect.objectContaining({
+          id: expect.any(Number),
+          title: expect.any(String),
+          userId: expect.any(Number),
+          description: expect.any(String),
+        }),
+      );
+
+      await db.product.delete({ where: { id } });
```

Comment:
Why not having this in an after each as you did with users?

Reply:
Because I don't really need this in all my unit tests. Actually in my last commit I removed all interaction with database because unit tests should not use database. I created a new file "product.services.new.integration.test.ts" where I moved some of these tests.